pytools/py_codegen.py

```diff
@@ -144,10 +145,14 @@ def __init__(self, module, name):
     def _initialize(self, module, name):
         self.module = module
         self.name = name
-        self.func = module.mod_globals[name]
+        if sys.version_info < (3, 9):
+            self.func = module.mod_globals[name]
+        else:
+            self.__call__ = module.mod_globals[name]
 
-    def __call__(self, *args, **kwargs):
-        return self.func(*args, **kwargs)
+    if sys.version_info < (3, 8):
+        def __call__(self, *args, **kwargs):  # pylint: disable=method-hidden
+            return self.func(*args, **kwargs)
```

Comment:
```suggestion
if sys.version_info < (3, 8):
    def __call__(self, *args, **kwargs):  # pylint: disable=method-hidden
        return self.func(*args, **kwargs)
```
? Otherwise that `__call__` doesn't seem to be doing anything?

Reply:
No, that's intentional. The `__call__` method definition should only run on Py3.7 and below. (Or at least that's the thought...)

Reply:
Yes, but it's only defined in the `_initialize`, not at the class level, right? (that's why I unindented the whole block)

Might be looking at it weird through the diff..

Reply:
Yep, looked at it weird, nevermind!